codex-rs/core/src/default_client.rs

```diff
@@ -134,22 +135,134 @@ pub fn create_client() -> CodexHttpClient {
     CodexHttpClient::new(inner)
 }
 
+/// Create an HTTP client with optional TLS configuration.
+/// Optionally configure TLS/mTLS settings via the `tls_config` parameter.
+pub fn create_configured_client(tls_config: Option<&TlsConfig>) -> CodexHttpClient {
+    let inner = build_configured_reqwest_client(tls_config);
+    CodexHttpClient::new(inner)
+}
+
 pub fn build_reqwest_client() -> reqwest::Client {
+    build_configured_reqwest_client(None)
+}
+
+pub fn build_configured_reqwest_client(tls_config: Option<&TlsConfig>) -> reqwest::Client {
+    let builder = create_base_client_builder();
+
+    // Apply TLS configuration if provided
+    let builder = if let Some(tls) = tls_config {
+        match apply_tls_config(builder, tls) {
+            Ok(configured_builder) => configured_builder,
+            Err(e) => {
+                tracing::error!("Failed to apply TLS configuration: {}", e);
+                // Fall back to base builder without TLS
+                create_base_client_builder()
+            }
+        }
+    } else {
+        builder
+    };
+
+    builder.build().unwrap_or_else(|_| reqwest::Client::new())
+}
+
+/// Create the base HTTP client builder with standard configuration.
+fn create_base_client_builder() -> reqwest::ClientBuilder {
     use reqwest::header::HeaderMap;
 
     let mut headers = HeaderMap::new();
     headers.insert("originator", originator().header_value.clone());
     let ua = get_codex_user_agent();
 
     let mut builder = reqwest::Client::builder()
-        // Set UA via dedicated helper to avoid header validation pitfalls
         .user_agent(ua)
         .default_headers(headers);
+
     if is_sandboxed() {
         builder = builder.no_proxy();
     }
 
-    builder.build().unwrap_or_else(|_| reqwest::Client::new())
+    builder
+}
+
+/// Apply TLS configuration to a reqwest ClientBuilder.
+fn apply_tls_config(
+    mut builder: reqwest::ClientBuilder,
+    tls: &TlsConfig,
+) -> Result<reqwest::ClientBuilder, String> {
+    use reqwest::Certificate;
+    use reqwest::Identity;
+
+    // Add custom CA certificate if provided
+    if let Some(ca_path) = &tls.ca_certificate {
+        let cert_pem = std::fs::read(ca_path).map_err(|e| {
+            format!(
+                "Failed to read CA certificate from {}: {}",
+                ca_path.display(),
+                e
+            )
```

Comment:
I feel like our clippy rules should catch this?

```suggestion
            format!(
                "Failed to read CA certificate from {}: {e}",
                ca_path.display(),
            )
```